### DISCUSSION TITLE
Where is the file to modify in order to change the max value of batch size and batch count?

### DISCUSSION THREAD

**Discussion by StableInfo** [vladmandic/sdnext#2093] [Q&A] [answered]
I can't find it.

**Comment by brknsoul** [comment]
Open `ui-config.json` with a text editor.
Search for "batch".
Locate or add `"txt2img/Batch count/maximum": 100,`
and `"txt2img/Batch size/maximum": 32,`
and adjust the numbers.

**Comment by StableInfo** [comment]
No, I already opened that file and found only 2 chunks of lines, none of them had the txt2img batch values, look:
![image](https://github.com/vladmandic/automatic/assets/136593585/dcd7db30-c601-422b-869b-8727aab6921b)

**Comment by vladmandic** [comment]
Given info is correct.
Values that are at defaults are not written to UI-config, but if you change them they are applied.

**Comment by StableInfo** [comment]
So instead of 'adjusting', I need to actually write new lines.
@brknsoul , you must be using an older version.

**Reply by brknsoul** [reply]
No, I've _updated_ from an older version.

**Comment by StableInfo** [comment, marked as answer]
> No, I've _updated_ from an older version.

my ui config file is as described,  I had to ADD he line you suggseted for me to modify in order to be able to modify them and it worked:
![image](https://github.com/vladmandic/automatic/assets/136593585/dc949179-cd37-4f5b-89fa-3d6ee25d0cc6)

**Reply by brknsoul** [reply]
Hence why I edited my response. ;-)

**Comment by Woisek** [comment]
A bit related to this:
The check boxes for Batch, Seed details, and Advanced are always unchecked which is annoying. Can this permanently checked, somewhere? 🤔

**Reply by brknsoul** [reply]
Yes. https://github.com/vladmandic/automatic/discussions/2084#discussioncomment-6833305

**Reply by Woisek** [reply]
Thanks, but I don't want to save the whole thing, I just want to have the three options permanently active/checked.

**Reply by vladmandic** [reply]
That's exactly it. Check them, go to UI settings and save that state.

**Reply by Woisek** [reply]
And everything I forgot to reset is saved with it like I had it a few times already. Can we have this a bit practically, please?
Is the ui-config.json the file where it is saved?

**Reply by vladmandic** [reply]
Yes.
Everything else? Let's be real, you'd do this once, not every 5min. Faster to do it than to type this message.

**Comment by StableInfo** [comment]
> Yes. [#2084 (comment)](https://github.com/vladmandic/automatic/discussions/2084#discussioncomment-6833305)

I think Vlad has probably a normal use of the UI, if we wanted simple parameters and simple UI (with everythinguncheked as he assumed) then we would have went to Midjouney or Foocus. A lot of users who know about sd.nextwant all parameters by defaults. IMO

**Comment by Woisek** [comment]
> Yes. Everything else? Let's be real, you'd do this once, not every 5min. Faster to do it than to type this message.

Let's be real? Then why do you assume that this is used 'only once'?

I don't want to start a discussion here, but it shows once again how rashly people act here.
Because you do not distinguish between UI display and value display. The two have almost nothing to do with each other. And you have them but thrown together, for whatever reason I will not say here. It is simply a bad work when UI settings and value settings are stored together. So now the problem is, that if someone starts to 'save', it can end in a vicious circle. 
And no, the reset is not really the solution, but only the last desperate possibility to delete everything you have already set so far and then have to set everything again.
And just with this Gradio shit, which does not delete values but preserves, this your solution to save all is the worst conceivable. More correct is to save UI and values separately.
But so ... 'one to save it all' is a bit hard. If you would have made it at least as a possibility for selectable presets ...

Just think, if Photoshop would really save everything when saving _all_ the preferences ... Brush size, colors, patterns, shapes, sizes, etc ...
But they do it separately. Guess why?
We see here very clearly how this dumb down thought influences the development. Unfortunately for the worse, it has to be said. It is happening as I predicted.
But what do I know ...

**Comment by StableInfo** [comment]
Isnt there an EXTENSION that save "state" of the UI @Woisek ? Maybe that can suit you better?

**Reply by Woisek** [reply]
@StableInfo I don't know, but it would be pretty weird if an extension is needed for something that is already built in, just not as well. It would make this implemented feature absurd, IMHO.

**Reply by StableInfo** [reply]
People here are trying their best, and are not always paid for their acrtins, we must accept than nothing can be perfect. And that people can be wrong smetimes and not see it, and that people are sometimes exhausted and fail to stay open minded and change their opinion. No big deal:).

**Reply by Woisek** [reply]
I'm absolutely with you, and I've said more than once how much I appreciate the work of all the developers here, because I know that path too. I just find it sad to see something go down the drain (I'm clearly exaggerating for a dramatic effect, just in case no one noticed) even though it was hinted at before.
But well, it is not my decision and I do not have to deal with it. 🙂

**Comment by vladmandic** [comment]
@Woisek some of the assumptions you draw are plain not correct (actually most of them in the last message are just wrong) and from there it just spirals downhill.

I suggest to start from your train of though from scratch (and that's giving you a pass - if you want to get the message accross, try to be constructive as this tone is not going to get anywhwre).

**Reply by StableInfo** [reply]
Can you @vladmandic  make a pole to ask for users opinions? This way we will know whats true nd whats not true, no one needs to assume anything or use theword "most" etc.

**Comment by Woisek** [comment]
@vladmandic I guess no one here has fact based informations, so every one 'assumes' something. I assume for example, that you are mainly a developer and not a (heavy) active user of your own product like I am. Therefore, I assume I can give more valid informations as anyone else not so dedicated. So my indent was to help and suggest as best I can, because, like I said many times, I know myself as developer that we often can miss even simplest things when we are deep in a project.
I also assume that you are not browse much SD relevant webpages, but I came not across just one post where users have already complained about how illogical and confusing the SD:Next interface has become. And often, it was something I suggested earlier on.

I don't know if you really want to discuss this out here, but if so, maybe if you point out where exactly I am wrong so I can correct myself, that would be great.

You maybe noticed that I don't involve anymore so much, because of such things. I would like to help, but somehow it is always seen as a personal attack, even you always write you are open for ideas/suggestions/other ways. At least that's how it comes across.
So I will continue to make my own tedious tweaks and if it becomes too much for me, then I will just switch to something else. And this thought will probably not only have me, that should also be clear. Or rather, I 'assume' this again ...

So, enough off topic from me, I'm out of this thread, I don't like where this is going and how this could escalate.
Nevertheless again, thanks for doing this project, vlad.

**Reply by StableInfo** [reply]
Hey @Woisek .  I share your opinion on some things (mainly I dont like some changes, such as the hires (second pass) box being far from the hires bars and values etc, and as I user like you I am describing actual feedback from experence).
However, I found a nice solution:), I made this post: https://github.com/vladmandic/automatic/issues/2098#issuecomment-1697240198
Interesting isnt it.

**Reply by Woisek** [reply]
This is actually a pretty good idea, and I'm glad Vlad accepted it. Maybe it should have been considered much earlier, but still, I think it's a useful idea, even if it's being implemented only now. 👍

**Comment by StableInfo** [comment]
Hello @Woisek and @vladmandic ,Woisek might be right on some things VLad,
For example, I dont understand the change on limiting the steps to 100 down from 150?
Here is a proof of a change that has  been made that is unlikeable by some users:
Check this page: https://civitai.com/models/15099/alfamix
This model states that you need 120-150 steps to get good results.
Yet you cannot  get more than 99 nowadays..
We really need the  feedback tab asap, or some thread for this

**Reply by vladmandic** [reply]
steps above 99 simply do not exist - they are either duplicates of previous steps or rounded down to zero due to math errors.
to go into details how math inside schedulers works is too much for here, but search for `timesteps` as i've spoken about it before.
also, why post this on a completely unrelated thread?

**Reply by Woisek** [reply]
@StableInfo Hi! I wouldn't give very much on such infos. Yes, the creator should know what settings his model is good at, but really ... such numbers should be a red flag. Like vlad mentioned, there are only so much of amount you can get, above it is either homeopathic or simply don't do anything. Some settings just doesn't make sense when set higher. Like the samples for instance.

**Comment by vladmandic** [comment]
> creator should know what settings his model is good at

also note that in many cases, creators are artists without deep knowledge of what's going on in the backend. so he may actually think he gets good results at setps 120-150, but that doesn't mean its true.

**Reply by Woisek** [reply]
Yes, that's totally true. That's why I'm happy to have some technical background for such things or gather myself some. 🙂

**Reply by StableInfo** [reply]
I speak with the creator of that model from times to times, he is adamant on having 150 steps (even on hires) to obtain the best rsults, he has been making models since long time, he spend all his time doing that it seems, I would trust him. In any case the resultst are INDEED different if you have 99 or 150 steps. I tried it before.
Its not that unrelated, I was refering to the comment woisek made about changes makers dont always know what their users are dong (using more than 99 steps for instance)

**Reply by Woisek** [reply]
> I speak with the creator of that model from times to times, he is adamant on having 150 steps (even on hires) to obtain the best rsults,

I don't want to say that it is totally impossible, but highly unlikely (for getting really noticeable better results).
But if we can see some high res sample images to compare, that would help. 🙂 Even better with the exact settings, so we could test this on our own.

**Reply by vladmandic** [reply]
What I said is that steps above 99 lead to unpredictable math such as duplicate steps or null steps or even reduced steps. 

For example, depending on scheduler, steps 150 may mean run 99 steps correctly and then run 51 steps with random values. Or run 99 steps correctly and then run 51 steps with null values. 

So is steps 150 different than 99? Yes.
Is it correct? Absolutely no.

**Comment by StableInfo** [comment]
Question: what do I need to add on the file to modify the max steps allowed? "txt2img/..???../maximum": 150,

**Reply by Woisek** [reply]
Not directly an answer, but a temporary possibility:
Open the input in question with the developer tools of the browser and override the max value of it. As long you don't refresh the page it stays that way.
